orttraining/orttraining/core/framework/gradient_graph_builder.cc

```diff
@@ -255,40 +256,54 @@ Status GradientGraphBuilder::Build(const std::unordered_set<std::string>* p_init
 
   // so far, visited are the minimum node in between
   // visited_node_args are the node_args involved
-  for (auto node : visited) {
-    //TODO: might not need two sets, the union of them might be enough
-    std::unordered_set<std::string> input_args_need_grad, output_args_need_grad;
-    for (auto arg : node->InputDefs()) {
-      if (visited_node_args.find(arg) != visited_node_args.end()) {
-        input_args_need_grad.insert(arg->Name());
+  std::unordered_set<std::string> stashed_tensors;
+  std::unordered_set<std::string> invertible_ops{"LayerNormalization"};
```

Comment:
>invertible_ops [](http://example.com/codeflow?start=34&length=14)

Move to gradient_graph_builder.h file? 
make is static const, just like STOP_GRADIENT_EDGES